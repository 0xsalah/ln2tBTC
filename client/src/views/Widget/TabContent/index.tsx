import React from 'react'

import { SwapPane, OperatePane, InvoicePane } from './Panes'


interface TabContentProps {
    activeTabId: string
}

const TabContent: React.FC<TabContentProps> = ({ activeTabId }) => {


    return (
        <div className="tab-content">
            {
                (activeTabId === 'swap' && <SwapPane />)
                || (activeTabId === 'operate' && <OperatePane />)
                || (activeTabId === 'invoice' && <InvoicePane />)
                || <div>Contact us.</div>
            }
        </div>
    )
}

export default TabContent