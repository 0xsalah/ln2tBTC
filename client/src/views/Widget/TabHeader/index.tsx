import React, { useState } from 'react'


interface TabHeaderProps {
    defaultTab: string
    onTabClick: (tabId: string) => void
}

const TabHeader: React.FC<TabHeaderProps> = ({ onTabClick, defaultTab }) => {
    const [currentTabId, setCurrentTabId] = useState(defaultTab)

    const handleTabClick = (e: React.MouseEvent<HTMLLIElement>) => {
        const id = e.currentTarget.getAttribute('data-tab-id') ?? defaultTab
        setCurrentTabId(id)
        onTabClick(id)
    }

    return (
        <ul className="tab-nav" data-tabs-nav="">
            <li onClick={handleTabClick} aria-label='swap' data-tab-id="swap" className={currentTabId === 'swap' ? 'is_active' : ''} ><span>Swap</span></li>
            <li onClick={handleTabClick} aria-label='operate' data-tab-id="operate" className={currentTabId === 'operate' ? 'is_active' : ''} ><span>Operate</span></li>
        </ul>
    )
}

export default TabHeader