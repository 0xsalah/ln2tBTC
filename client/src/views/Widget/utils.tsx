export const toMaxDecimalsFloor = (n: number | string, step: number) => {
    const amount = typeof n === 'string' ? Number(n.replace(',', '.')) : n
    //let factor = Number('1e' + decimals)
    let nRound = Math.floor((amount) * step) / step
    return !isFinite(nRound) ? 0 : nRound
}

export const toMaxDecimalsRound = (n: number | string, step: number) => {
    const amount = typeof n === 'string' ? Number(n.replace(',', '.')) : n
    if (step <= 0) return amount
    //let factor = Number('1e' + decimals)
    let factor = (1 / step)
    let nRound = Math.round((amount) * factor) / factor
    console.log('nroud', step === 1, step, (1 / step))
    return !isFinite(nRound) ? 0 : nRound
}