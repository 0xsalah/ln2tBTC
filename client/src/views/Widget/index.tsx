import React, { useState } from 'react'
import TabHeader from './TabHeader'
import TabContent from './TabContent'

interface WidgetProps {
}

const Widget: React.FC<WidgetProps> = () => {
    const defaultTab = 'swap'
    const [activeTabId, setActiveTabId] = useState(defaultTab)

    return (
        <div className="box-operation">
            <div className='box-operation__tabs tabs'>
                <TabHeader defaultTab={defaultTab} onTabClick={setActiveTabId} />
                <TabContent activeTabId={activeTabId} />
            </div>
        </div>
    )
}

export default Widget