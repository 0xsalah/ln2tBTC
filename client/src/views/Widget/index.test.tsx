import React from "react";
import { render } from "@testing-library/react";
import Widget from ".";

test("snapshot of component rendered", () => {
  const component = render(<Widget />);
  expect(component.baseElement).toMatchSnapshot();
});
