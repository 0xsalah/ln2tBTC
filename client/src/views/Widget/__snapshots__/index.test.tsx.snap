// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of component rendered 1`] = `
<body>
  <div>
    <div
      class="box-operation"
    >
      <div
        class="box-operation__tabs tabs"
      >
        <ul
          class="tab-nav"
          data-tabs-nav=""
        >
          <li
            aria-label="swap"
            class="is_active"
            data-tab-id="swap"
          >
            <span>
              Swap
            </span>
          </li>
          <li
            aria-label="operate"
            class=""
            data-tab-id="operate"
          >
            <span>
              Operate
            </span>
          </li>
        </ul>
        <div
          class="tab-content"
        >
          <div
            class="tab-pane is_active"
          >
            <div
              class="tab-pane__content"
            >
              <div
                class="box-operation__content box-operation__content--swap"
              >
                <div
                  class="box-operation__exchange exchange"
                >
                  <div
                    class="exchange__form"
                  >
                    <div
                      class="exchange__row row align-end"
                    >
                      <div
                        class="exchange__column exchange__column--from"
                      >
                        <div
                          class="form-block "
                        >
                          <label
                            class="form-block__label"
                          >
                            <span>
                              From
                            </span>
                          </label>
                          <div
                            class="form-block__box"
                          >
                            <input
                              aria-label="input"
                              class="form-block__input form-control"
                              min="0"
                              name="tbtc"
                              placeholder="0.0"
                              step="0.001"
                              type="number"
                              value=""
                            />
                            <div
                              class="form-block__unit action-text"
                            >
                              MAX
                            </div>
                            <div
                              class="form-block__icon"
                            >
                              <svg
                                class="icon icon-man no-fill-transition"
                              >
                                <use
                                  xlink:href="icons.svg#icon-man"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="tooltipster-base tooltipster-sidetip tooltipster-0 fff tooltipster-fade tooltipster-dying"
                        style="position: absolute; max-width: 0px; max-height: 0px; z-index: 9999999; top: 0px; left: 0px; animation-duration: 350ms; transition-duration: 350ms;"
                      >
                        <div
                          class="tooltipster-box"
                        >
                          <div
                            class="tooltipster-content"
                          />
                        </div>
                        <div
                          class="tooltipster-arrow"
                          style="left: 0px;"
                        >
                          <div
                            class="tooltipster-arrow-uncropped"
                          >
                            <div
                              class="tooltipster-arrow-border"
                            />
                            <div
                              class="tooltipster-arrow-background"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="exchange__column exchange__column--icon"
                      >
                        <div
                          class="exchange__icon"
                        >
                          <svg
                            class="icon icon-direction"
                          >
                            <use
                              xlink:href="icons.svg#icon-direction"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        class="exchange__column exchange__column--to"
                      >
                        <div
                          class="form-block "
                        >
                          <label
                            class="form-block__label"
                          >
                            <span>
                              To
                            </span>
                          </label>
                          <div
                            class="form-block__box"
                          >
                            <input
                              aria-label="input"
                              class="form-block__input form-control"
                              min="0"
                              name="ln"
                              placeholder="0.0"
                              step="0.001"
                              type="number"
                              value=""
                            />
                            <div
                              class="form-block__icon"
                            >
                              <svg
                                class="icon icon-flash no-fill-transition"
                              >
                                <use
                                  xlink:href="icons.svg#icon-flash"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="tooltipster-base tooltipster-sidetip tooltipster-0 fff tooltipster-fade tooltipster-dying"
                        style="position: absolute; max-width: 0px; max-height: 0px; z-index: 9999999; top: 0px; left: 0px; animation-duration: 350ms; transition-duration: 350ms;"
                      >
                        <div
                          class="tooltipster-box"
                        >
                          <div
                            class="tooltipster-content"
                          />
                        </div>
                        <div
                          class="tooltipster-arrow"
                          style="left: 0px;"
                        >
                          <div
                            class="tooltipster-arrow-uncropped"
                          >
                            <div
                              class="tooltipster-arrow-border"
                            />
                            <div
                              class="tooltipster-arrow-background"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="exchange__button"
                    >
                      <button
                        class="btn btn--full btn--secondary"
                      >
                        <span>
                          Connect wallet
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
